File: Data Structures & Algorithms/two-integer-sum/submission-12.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }

        int[] ans = new int[2];

        for(int i = 0; i < nums.length; i++) {
            int comp = target - nums[i];
            if(map.containsKey(comp) && map.get(comp) != i) {
                ans[0] = map.get(comp);
                ans[1] = i;
                break;
            }
            
        }
        Arrays.sort(ans);
        return ans;
    }
}

File: Data Structures & Algorithms/binary-search/submission-12.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        m = len(nums) // 2
        r = len(nums) - 1
        for i in range(0, len(nums)//2 + 1):
            if(nums[m] == target):
                 return m
            elif(nums[m] > target):
                r = m - 1
                m = (l + r) // 2
            else:
                l = m + 1
                m = (l + r) // 2
        return -1
